src/test.js

```diff
@@ -1,7 +1,324 @@
+      wrapperHeight: 250,
+      holePositionY: (300 - 150) / 2,
+      holePositionX: (700 - 150) / 2,
+    })
+    component.instance()._calculateDraggableBounds()
+  })
+
+  describe('hole', () => {
+    describe('when component did mount', () => {
+      describe('when hole position is center', () => {
+        test('calculate hole position', () => {})
+      })
+    })
+
+    describe('when picture position X change', () => {
+      beforeEach(() => {
+        onChange.mockClear()
+        const draggableComponent = component.find('Draggable')
+        draggableComponent.props().onStop({}, { x: 50, y: 0 })
+      })
+
+      test('submit change to parent', () => {
+        expect(onChange).toBeCalledWith(expect.objectContaining({
+          x: 225,
+          y: 75,
+        }))
+      })
+    })
+
+    describe('when picture position Y change', () => {
+      beforeEach(() => {
+        onChange.mockClear()
+        const draggableComponent = component.find('Draggable')
+        draggableComponent.props().onStop({}, { x: 0, y: 50 })
+      })
+
+      test('submit change to parent', () => {
+        expect(onChange).toBeCalledWith(expect.objectContaining({
+          x: 275,
+          y: 25,
+        }))
+      })
+    })
+  })
+
+  describe('hole bounds', () => {
+    describe('when rotate 0deg & zoom x 2', () => {
+      beforeEach(() => {
+        component.setState({
+          currentZoom: 2,
+          currentRotation: 0,
+        })
+      })
+
+      test('correctly calculate bounds', () => {
+        expect(component.state('bounds')).toMatchObject({
+          bottom: 75,
+          left: -975,
+          right: 275,
+          top: -375,
+        })
+      })
+    })
+    describe('when rotate 90deg & zoom x 1', () => {
+      beforeEach(() => {
+        component.instance().setState({
+          currentRotation: 90,
+          currentZoom: 1,
+        })
+      })
+
+      test('correctly calculate bounds', () => {
+        expect(component.state('bounds')).toMatchObject({
+          bottom: 75,
+          left: 125,
+          right: 275,
+          top: -475,
+        })
+      })
+    })
+    describe('when rotate 90deg & zoom x 2', () => {
+      beforeEach(() => {
+        component.instance().setState({
+          currentRotation: 90,
+          currentZoom: 2,
+        })
+      })
+
+      test('correctly calculate bounds', () => {
+        expect(component.state('bounds')).toMatchObject({
+          bottom: 75,
+          left: -175,
+          right: 275,
+          top: -1175,
+        })
+      })
+    })
+  })
+
+  describe('with zoom actions', () => {
+    let beforeProps = {}
+    beforeAll(() => {
+      beforeProps = {...props}
+      props = {
+        enableZoomActions: true,
+      }
+    })
+    afterAll(() => {
+      props = {...beforeProps}
+    })
+
+    test('display zoom controls', () => {
+      expect(component.text()).toBe('+-')
+    })
+
+    describe('when zoom in', () => {
```

Comment:
```suggestion
     describe('when zooming in', () => {
```

Et là je note que tu as bien appelé ça zoom In ;-)